llama_stack/providers/utils/vector_io/chunk_utils.py

```diff
@@ -9,6 +9,18 @@
 
 
 def generate_chunk_id(document_id: str, chunk_text: str) -> str:
-    """Generate a unique chunk ID using a hash of document ID and chunk text."""
+    """
+    Generate a unique chunk ID using a hash of the document ID and chunk text.
+
+    Note: MD5 is used only to calculate an identifier, not for security purposes.
+    Adding usedforsecurity=False for compatibility with FIPS environments.
+    """
     hash_input = f"{document_id}:{chunk_text}".encode()
-    return str(uuid.UUID(hashlib.md5(hash_input).hexdigest()))
+
+    try:
+        md5_hash = hashlib.md5(hash_input, usedforsecurity=False).hexdigest()
+    except TypeError:
```

Comment:
I don't think we need this fallback, the project supports Python >= 3.12. Also, if Python was not compiled with OpenSSL I believe `usedforsecurity` is a noop.